src/print/index.ts

```diff
@@ -530,174 +526,152 @@ function printAttributeNodeValue(
     }
 }
 
-function printChildren(path: FastPath, print: PrintFn): Doc[] {
-    let childDocs: Doc[] = [];
-    let currentGroup: { doc: Doc; node: Node }[] = [];
-    // the index of the last child doc we could add a linebreak after
-    let lastBreakIndex = -1;
-
-    const isPreformat = isPreTagContent(path);
-
-    /**
-     * Call when reaching a point where a linebreak is possible. Will
-     * put all `childDocs` since the last possible linebreak position
-     * into a `concat` to avoid them breaking.
-     */
-    function linebreakPossible() {
-        if (lastBreakIndex >= 0 && lastBreakIndex < childDocs.length - 1) {
-            childDocs = childDocs
-                .slice(0, lastBreakIndex)
-                .concat(concat(childDocs.slice(lastBreakIndex)));
-        }
-
-        lastBreakIndex = -1;
+function printChildren2(
```

Comment:
maybe this could be broken up into three or four functions since it's essentially calculating the flags followed by three independent cases?